var/spack/repos/builtin/packages/tmux/package.py

```diff
@@ -30,6 +30,7 @@ class Tmux(AutotoolsPackage):
     version('1.9a', sha256='c5e3b22b901cf109b20dab54a4a651f0471abd1f79f6039d79b250d21c2733f5')
 
     depends_on('libevent')
+    # assumes that the default is ncurses+termlib, see configure_args below
```

Comment:
This isn't a good assumption. You should check `if '+termlib' in self.spec['ncurses']` to confirm.

Reply:
Fair enough.  What should I do if it's not?  Fail?

An alternative would be to change to `depends_on('ncurses+termlib')` and be done with it.

Reply:
That depends, can `tmux` build with `ncurses~termlib`? If not, update `depends_on`. If it requires a different linking flag, use an if-else statement.

Reply:
I took a different approach.  `LIBTINFOLIBS` is defined for overriding pkg-config.  Our pkg-config seems to do the right thing for our curses library, so I changed the depends on to build curses with termlib, then changed the pkg definition so that it relies on pkg-config.

I've updated the PR summary.

That leaves me with the question of whether I need an explicit depends_on('pkg-config').

Looks like I'm getting it "for free" via ncurses.

```
$ spack spec tmux
Input spec
--------------------------------
tmux

Concretized
--------------------------------
tmux@3.0a%gcc@8.2.0 arch=linux-centos7-skylake_avx512
    ^libevent@2.1.8%gcc@8.2.0+openssl arch=linux-centos7-skylake_avx512
        ^openssl@1.1.1d%gcc@8.2.0+systemcerts arch=linux-centos7-skylake_avx512
            ^perl@5.26.2%gcc@8.2.0+cpanm patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac +shared+threads arch=linux-centos7-skylake_avx512
                ^gdbm@1.18.1%gcc@8.2.0 arch=linux-centos7-skylake_avx512
                    ^readline@8.0%gcc@8.2.0 arch=linux-centos7-skylake_avx512
                        ^ncurses@6.1%gcc@8.2.0~symlinks+termlib arch=linux-centos7-skylake_avx512
                            ^pkgconf@1.6.3%gcc@8.2.0 arch=linux-centos7-skylake_avx512
            ^zlib@1.2.11%gcc@8.2.0+optimize+pic+shared arch=linux-centos7-skylake_avx512

```

Thoughts?

Reply:
Yes, you should have an explicit dependency on pkg-config

Reply:
Done, looks like it's spelled 'pkgconfig' in many, many other packages.

Reply:
I wonder if this would fix the Cray problem that led to the `flag_handler` bit?    I can ping the author, but would prefer not to hang up the PR as is.

Reply:
> Done, looks like it's spelled 'pkgconfig' in many, many other packages.

It's a virtual dependency, provided by both `pkg-config` and `pkgconf` (the default).



> I wonder if this would fix the Cray problem that led to the `flag_handler` bit? I can ping the author, but would prefer not to hang up the PR as is.

Probably. Try removing it and make sure it builds properly? I don't use `tmux`, and there's no `maintainer` listed, so if it breaks and someone is upset, then we've found our maintainer 😄

Reply:
Removed it.  Works for me (I can build all of the versions the tmux package defines and ldd shows all linking to our libtinfo).

That bit landed [in this commit](https://github.com/spack/spack/commit/407834fffa9b9b247ecc4dee7888231079325b04), arguably if it's cray specific it should have been protected via a platform check.

I can mention the author and have him check sooner rather than later, but it'd be nice if this landed so I can get on with it.

Reply:
> we've found our maintainer 😄

That'd be Not Me, as I'm not getting paid to do this stuff any more (just helping out a friend).